main.go

```diff
@@ -87,8 +87,8 @@ func parseFlags() (*config, error) {
 	rawURL := flag.String("url", "", "The target URL to which to proxy requests. All requests will have the acces token in the Authorization HTTP header.(DEPRECATED: Use -upstream.url instead)")
 	rawUpstreamURL := flag.String("upstream.url", "", "The target URL to which to proxy requests. All requests will have the acces token in the Authorization HTTP header.")
 	flag.StringVar(&cfg.upstream.caFile, "upstream.ca-file", "", "The path to the CA file to verify upstream server TLS certificates.")
-	flag.DurationVar(&cfg.upstream.readTimeout, "upstream.read-timout", 5*time.Second, "The time from when the connection is accepted to when the request body is fully read.")
-	flag.DurationVar(&cfg.upstream.writeTimeout, "upstream.write-timout", 10*time.Second, "The time from the end of the request header read to the end of the response write .")
+	rawReadTimeout := flag.String("upstream.read-timeout", "", "The time from when the connection is accepted to when the request body is fully read.")
```

Comment:
This change seems like a downgrade in functionality. Now users don't know that the flag should of type duration and can take time suffixes and we have to do manual string parsing. Why not keep it as a duration with default 0 so the timeouts are disabled?

Reply:
Good catch, I was misguided from the notion `time.Duration` is a custom type and has no primitive default value but it's a simple int64 :face_exhaling: